### PR TITLE
Additional copy option

### PR DESCRIPTION
**Pull request by ColinScott**

### CODE REVIEW

File: LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        public void CanCopyBlob(string source, string destination)
+        {
+            var buffer = Encoding.UTF8.GetBytes("File content");
+            var sourceBlob = new StandaloneAzureBlockBlob(BasePath, source);
+            sourceBlob.UploadFromByteArrayAsync(buffer, 0, buffer.Length).Wait();
+
+            var destinationBlob = new StandaloneAzureBlockBlob(BasePath, destination);
+            destinationBlob.StartCopyFromBlob(sourceBlob.Uri);
+
+            Assert.Equal("File content", File.ReadAllText(destinationBlob.Uri.LocalPath));
```
Comment (cc-zulhafizakbar) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:49]: Why not `Assert.Equal( File.ReadAllText(sourceBlob.Uri.LocalPath, File.ReadAllText(destinationBlob.Uri.LocalPath));`?

Comment (ColinScott) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:49]: More complex given the text is static. Probably worth doing if the string is randomised (which would be a nice enhancement)

File: LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        {}
+
+        public static IEnumerable<object[]> CopyBlobNames
+        {
+            get
+            {
+                yield return new object[] { "source", "destination" };
+                yield return new object[] { "source", @"destination\with\path" };
+                yield return new object[] { "source", "destination/with/path/alternate" };
+                yield return new object[] { @"source\with\path", "destination" };
```
Comment (cc-zulhafizakbar) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:30]: `yield return new object[] { @"source\with\path", "destination/with/path/alternate" };`

File: LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                yield return new object[] { "source/with/path/alternate", "destination" };
```
Comment (cc-zulhafizakbar) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:31]: `yield return new object[] { "source/with/path/alternate", " @"destination\with\path" };`
`yield return new object[] { "source/with/path/alternate", " "destination/with/path/alternate" };`

File: LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                yield return new object[] { @"source\with\path", @"destination\with\path" };
+            }
+        }
+
+        [Theory]
+        [PropertyData("CopyBlobNames")]
+        public void CanCopyBlob(string source, string destination)
+        {
+            var buffer = Encoding.UTF8.GetBytes("File content");
```
Comment (cc-zulhafizakbar) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:42]: Does it matter if encoding is not UTF8?

File: LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+            sourceBlob.Metadata["thing"] = "something";
+            sourceBlob.Properties.ContentType = "whatever";
+            sourceBlob.SetMetadata();
+            sourceBlob.SetProperties();
+
+            var destinationBlob = new StandaloneAzureBlockBlob(BasePath, destination);
+            destinationBlob.StartCopyFromBlob(sourceBlob.Uri);
+            destinationBlob.FetchAttributes();
+
+            new
```
Comment (cc-zulhafizakbar) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:98]: Why not just test if the meta file is exist?

File: LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs (hunk 4 above)
Comment (ColinScott) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:42]: No, this just gets us bytes.

File: LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs (hunk 5 above)
Comment (ColinScott) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:98]: Makes the tests less aware of the underlying implementation.

Comment (cc-zulhafizakbar) [inline LightBlue.Tests/Standalone/StandaloneAzureBlockBlobCopyByUriTests.cs:98]: Sure, but you're testing `FetchAttributes` though

**Pull request merged**